socorro/signature/rules.py

```diff
@@ -109,89 +109,22 @@ def __init__(self, quit_check_callback=None):
-                return True
-            if line_up_to_current_position.endswith(an_exception):
-                return True
-        return False
-
-    def _collapse(
-        self,
-        function_signature_str,
-        open_string,
-        replacement_open_string,
-        close_string,
-        replacement_close_string,
-        exception_substring_list=None,
-    ):
-        """Replaces bits between two (possibly nested) delimiters
-
-        this method takes a string representing a C/C++ function signature and
-        replaces anything between two possibly nested delimiters
-
-        :arg str function_signature_str: the original string
-        :arg str open_string: the open string. e.g. "<"
-        :arg str replacement_open_string: the string to replace the open string with
-        :arg str close_string: the close string. e.g. ">"
-        :arg str replacement_close_string: the string to replace the closestring with
-        :arg list exception_substring_list: list of exceptions that shouldn't collapse
-
-        """
-        target_counter = 0
-        collapsed_list = []
-        exception_mode = False
-        exception_substring_list = exception_substring_list or []
-
-        def append_if_not_in_collapse_mode(a_character):
-            if not target_counter:
-                collapsed_list.append(a_character)
-
-        for index, a_character in enumerate(function_signature_str):
-            if a_character == open_string:
-                if self._is_exception(
-                    exception_substring_list,
-                    function_signature_str[index + 1:],
-                    function_signature_str[:index]
-                ):
-                    exception_mode = True
-                    append_if_not_in_collapse_mode(a_character)
-                    continue
-                append_if_not_in_collapse_mode(replacement_open_string)
-                target_counter += 1
-            elif a_character == close_string:
-                if exception_mode:
-                    append_if_not_in_collapse_mode(a_character)
-                    exception_mode = False
-                else:
-                    target_counter -= 1
-                    append_if_not_in_collapse_mode(replacement_close_string)
-            else:
-                append_if_not_in_collapse_mode(a_character)
-
-        edited_function = ''.join(collapsed_list)
-        return edited_function
-
     def normalize_rust_function(self, function, line):
         """Normalizes a single rust frame with a function"""
-        function = self._collapse(
+        function = collapse(
             function,
             open_string='<',
-            replacement_open_string='<',
             close_string='>',
-            replacement_close_string='T>',
-            exception_substring_list=('name omitted', 'IPC::ParamTraits')
+            replacement='<T>',
+            exceptions=('name omitted', 'IPC::ParamTraits', ' as ')
```

Comment:
I should check if "IPC::" ever shows up in Rust frames. I'm pretty sure it doesn't. If so, then we could nix that. Plus we could probably nix the `name omitted` exception, too.